src/metadataGeneration/typeResolver.ts

```diff
@@ -832,7 +855,7 @@ export class TypeResolver {
       throw new GenerateMetadataError(`No valid type found for property declaration.`);
     }
 
-    const type = new TypeResolver(typeNode, this.current, propertyDeclaration, this.context).resolve();
+    const type = new TypeResolver(typeNode, this.current, propertyDeclaration, this.context, typeNode as ts.TypeReferenceType).resolve();
```

Comment:
Maybe we can check if it's a ref. If it is, we can set is without casting, if it is not we don't need it afaik.

Reply:
It seems that `typeNode` (in our case) is a `IndexedAccessType` instead of a `TypeReferenceType`, so if we can a `if` it don't work anymore

Reply:
This means we should either change our expectations of a `referencer` or find a good assertion function that makes sure whatever we pass here won't crash. As is, this will eventually crash.